src/OrleansRuntime/Hosting/SiloBuilderExtensions.cs

```diff
@@ -0,0 +1,94 @@
+        /// <remarks>This method may only be called once per builder instance.</remarks>
+        /// <returns>The silo builder.</returns>
+        public static ISiloBuilder UseConfiguration(this ISiloBuilder builder, ClusterConfiguration configuration)
+        {
+            return builder.ConfigureServices(services => services.AddSingleton(configuration));
+        }
+
+        /// <summary>
+        /// Loads <see cref="ClusterConfiguration"/> using <see cref="ClusterConfiguration.StandardLoad"/>.
+        /// </summary>
+        /// <param name="builder">The silo builder.</param>
+        /// <returns>The silo builder.</returns>
+        public static ISiloBuilder LoadClusterConfiguration(this ISiloBuilder builder)
+        {
+            var configuration = new ClusterConfiguration();
+            configuration.StandardLoad();
+            return builder.UseConfiguration(configuration);
+        }
+        
+        /// <summary>
+        /// Configures a localhost silo.
+        /// </summary>
+        /// <param name="builder">The silo builder.</param>
+        /// <param name="siloPort">The silo-to-silo communication port.</param>
+        /// <param name="gatewayPort">The client-to-silo communication port.</param>
+        /// <returns>The silo builder.</returns>
+        public static ISiloBuilder ConfigureLocalHostPrimarySilo(this ISiloBuilder builder, int siloPort = 22222, int gatewayPort = 40000)
+        {
+            builder.ConfigureSiloName(Silo.PrimarySiloName);
+            return builder.UseConfiguration(ClusterConfiguration.LocalhostPrimarySilo(siloPort, gatewayPort));
+        }
+
+        /// <summary>
+        /// Configures a startup class.
+        /// </summary>
+        /// <typeparam name="TStartup">The startup class, which must conform to the correct shape.</typeparam>
+        /// <param name="builder">The silo builder.</param>
+        /// <returns>The silo builder.</returns>
+        /// <remarks>
+        /// The provided <typeparamref name="TStartup"/> type must have a method matching this signature:
+        /// <code>
+        /// IServiceProvider ConfigureServices(IServiceCollection);
+        /// </code>
+        /// </remarks>
+        public static ISiloBuilder UseLegacyStartup<TStartup>(this ISiloBuilder builder) where TStartup : class
+        {
+            var configureServicesMethod = StartupBuilder.FindConfigureServicesDelegate(typeof(TStartup));
+            if (configureServicesMethod == null) throw new ArgumentException($"Type {typeof(TStartup)} does not have a suitable method for configuring a service provider.");
+            if (configureServicesMethod.MethodInfo.IsStatic) throw new ArgumentException($"Method {configureServicesMethod} on type {typeof(TStartup)} must not be static.");
+            return builder.ConfigureServiceProvider(svc => StartupBuilder.ConfigureStartup(typeof(TStartup).AssemblyQualifiedName, svc));
+        }
```

Comment:
Probably not, since I believe the plan is to deprecate it eventually. If the user can still configure the startup type via the cluster configuration, there's no need to add another API for it.
It is not very clear to me yet how do we execute/configure stuff after the service provider was created, but still not sure whether it will be needed once we have named options.